securedrop/tests/common.py

```diff
@@ -1,120 +1,291 @@
+# -*- coding: utf-8 -*-
+
+from functools import wraps
+import gnupg
+import mock
 import os
 import shutil
-import uuid
 import subprocess
-
-import gnupg
+import time
 
 # Set environment variable so config.py uses a test environment
 os.environ['SECUREDROP_ENV'] = 'test'
+
 import config
-from db import init_db, db_session, Journalist, Reply, Source, Submission
+from db import db_session, init_db, Journalist, Reply, Source, Submission
 import crypto_util
+import store
 
-# TODO: the PID file for the redis worker is hard-coded below.
-# Ideally this constant would be provided by a test harness.
-# It has been intentionally omitted from `config.py.example`
-# in order to isolate the test vars from prod vars.
-# When refactoring the test suite, the TEST_WORKER_PIDFILE
-# TEST_WORKER_PIDFILE is also hard-coded in `manage.py`.
-TEST_WORKER_PIDFILE = "/tmp/securedrop_test_worker.pid"
 
+class TestJournalist:
+    """Test fixtures for working with :class:`db.Journalist`.
+    """
+    @staticmethod
+    def init_journalist(is_admin=False):
+        """Initialize a journalist into the database. Return their
+        :class:`db.Journalist` object and password string.
 
-def clean_root():
-    shutil.rmtree(config.SECUREDROP_DATA_ROOT)
+        :param bool is_admin: Whether the user is an admin.
 
+        :returns: A 2-tuple. The first entry, the :class:`db.Journalist`
+                  initialized. The second, their password string.
+        """
+        username = crypto_util.genrandomid()
+        user_pw = crypto_util.genrandomid()
```

Comment:
Don't think it's a big deal in this case, but we can hard-code a seed with `random.seed()` in `common.SetUp.setup()`. I'd rather do that and make the API calls for everything rather than hardcoding in actual values in these cases, as it is a better way to test functions are working in conjunction with each other as expected. This was also, if any of the API calls we make change, we don't have to change every fixture such that it could be a state produced by calls to those functions.